engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java

```diff
@@ -2200,7 +2200,7 @@ public Network createGuestNetwork(final long networkOfferingId, final String nam
             }
             if (! UuidUtils.validateUUID(vlanId)){
                 // For Isolated and L2 networks, don't allow to create network with vlan that already exists in the zone
-                if (ntwkOff.getGuestType() == GuestType.Isolated || ntwkOff.getGuestType() == GuestType.L2) {
+                if (ntwkOff.getGuestType() == GuestType.Isolated || !(bypassVlanOverlapCheck && ntwkOff.getGuestType() == GuestType.L2)) {
```

Comment:
@gmueller-ewerk this conditional is growing. Can you please extract it to a method? With such a method, it would be possible to have a Javadoc and a couple of JUnit test methods asserting the expected behavior.

Reply:
Good suggestion